### ISSUE TITLE
Example in Getting-started doesn't work

### ISSUE THREAD

**Issue by Duskfall** [loedeman/AutoMapper#11]
I am using the automapper and it used to work nicely but not now it seems. I need the functionallity that you present in the "getting started guide", to add new properties to destination model.

If we run the code:

    var objA = { prop1: 'From A', prop2: 'From A too', prop3: 'Also from A (really)' };

    automapper
       .createMap('sourceType', 'destinationType')
       .forMember('prop1', function (opts) { opts.mapFrom('prop2'); })
       .forMember('prop2', function (opts) { opts.ignore(); })
       .forSourceMember('prop3', function (opts) { opts.ignore(); })
       .forMember('prop4', function () { return 12; })

    var objB = automapper.map('sourceType', 'destinationType', objA);

instead of your result we get

{prop1: "From A"} 

**and not **

{
  "prop1": "From A too", 
  "prop4": 12 
}

Also another problem is that if we have a nested object, which has another mapper and **has null value** we can't just use

````.forMember('item', (opts: AutoMapperJs.IMemberConfigurationOptions) => {
              automapper.map('ItemDto', 'ItemModel', opts.sourceObject.item)
  });````

because it returns an empty object but we want the null value.

I managed to bypass this by using

```
.forMember('item', (opts: AutoMapperJs.IMemberConfigurationOptions) => {
             return opts.sourceObject.item ? automapper.map('ItemDto', 'ItemModel', opts.sourceObject.item) : null
  });
```

**Comment by jrusinek** [comment]
I second the first issue. Adding new members doesn't work.

**Comment by loedeman** [comment]
Thanks for using AutoMapper and reportage those issues. I will have a look next week, on a vacation right now ;)...

**Comment by Duskfall** [comment]
Have fun!

**Comment by loedeman** [comment]
I checked the first one, the example in the getting started guide does no longer work indeed. Ignoring a member completely ignores it, even if it was mapped from another property before. However, this works just fine for the example:

```
var objA = { prop1: 'From A', prop2: 'From A too', prop3: 'Also from A (really)' };

var fromKey = 'should be able to use mapFrom to switch ';
var toKey = 'properties and ignore a property as well';

automapper
    .createMap(fromKey, toKey)
    .forMember('prop1', (opts: IMemberConfigurationOptions) => { opts.mapFrom('prop2'); })
    .forMember('prop2', (opts: IMemberConfigurationOptions) => { opts.mapFrom('prop1'); })
    .forSourceMember('prop3', (opts: ISourceMemberConfigurationOptions) => { opts.ignore(); });

var objB = automapper.map(fromKey, toKey, objA);

expect(objB).toEqualData({ prop1: objA.prop2, prop2: objA.prop1 });
```

The part of adding new members does not work either, I just fixed that one ;) ...

I will have a look at the 'null object' and create a new release after that.

**Comment by loedeman** [comment]
Hi @Duskfall @jrusinek , 

Just released v1.6.4 which should have solved your issues. Please let me know if it does not ;).

Cheers!
